src/language/templates/pybind/pyvisitor.cpp

```diff
@@ -187,13 +216,9 @@ void init_visitor_module(py::module& m) {
         .def(py::init<ast::AstNodeType>())
         .def("get_nodes", &AstLookupVisitor::get_nodes)
         .def("clear", &AstLookupVisitor::clear)
-        .def("lookup", (std::vector<std::shared_ptr<ast::Ast>> (AstLookupVisitor::*)(ast::Ast&)) &AstLookupVisitor::lookup)
-        .def("lookup", (std::vector<std::shared_ptr<ast::Ast>> (AstLookupVisitor::*)(ast::Ast&, ast::AstNodeType)) &AstLookupVisitor::lookup)
-        .def("lookup", (std::vector<std::shared_ptr<ast::Ast>> (AstLookupVisitor::*)(ast::Ast&, const std::vector<ast::AstNodeType>&)) &AstLookupVisitor::lookup)
-    {% for node in nodes %}
-        .def("visit_{{ node.class_name | snake_case }}", &AstLookupVisitor::visit_{{ node.class_name | snake_case }})
-        {% if loop.last -%};{% endif %}
-    {% endfor %}
```

Comment:
I removed this because IMO these are internal methods that should not be exposed in either C++ or Python public API of the AstLookupVisitor. Note that in the counterpart C++ class, this PR changes the visibility of the `visit_` methods to _protected_.

The Python `AstLookupVisitor.lookup` static methods directly call the C++ code, so the `visit_foo` Python methods are not used.

UTs of the Python AstLookupVisitor class still passes:
```
git grep -A2 AstLookupVisitor test/unit/pybind
test/unit/pybind/test_symtab.py:    lookup_visitor = visitor.AstLookupVisitor()
test/unit/pybind/test_symtab.py-    eqs = lookup_visitor.lookup(ch_ast, ast.AstNodeType.DIFF_EQ_EXPRESSION)
test/unit/pybind/test_symtab.py-    stream = io.StringIO()
--
test/unit/pybind/test_visitor.py:    lookup_visitor = visitor.AstLookupVisitor()
test/unit/pybind/test_visitor.py-    eqs = lookup_visitor.lookup(ch_ast, ast.AstNodeType.DIFF_EQ_EXPRESSION)
test/unit/pybind/test_visitor.py-    eq_str = nmodl.dsl.to_nmodl(eqs[0])
--
test/unit/pybind/test_visitor.py:    lookup_visitor = visitor.AstLookupVisitor(ast.AstNodeType.DIFF_EQ_EXPRESSION)
test/unit/pybind/test_visitor.py-    eqs = lookup_visitor.lookup(ch_ast)
test/unit/pybind/test_visitor.py-    eq_str = nmodl.dsl.to_nmodl(eqs[0])
--
test/unit/pybind/test_visitor.py:    lookup_visitor = visitor.AstLookupVisitor(ast.AstNodeType.PRIME_NAME)
test/unit/pybind/test_visitor.py-    primes = lookup_visitor.lookup(ch_ast)
test/unit/pybind/test_visitor.py-
```

These tests always pass `ch_ast` which is a `ast::Program` whereas the `AstLookupVisitor::lookup` signature consumes a `ast::Ast`. So it is fair to assume that you can pass any node. I added a UT to be sure though, see 31c67d8

Reply:
Removing these are not big deal but I was wondering if this affects anything when we `extend` visitor in Python as done [here](https://bluebrain.github.io/nmodl/html/notebooks/nmodl-python-tutorial.html#Easy-code-generation-using-AST-visitors).

Reply:
I ran the following test:

```
def test_my_custom_lookup_visitor():
    class MyAstLookupVisitor(visitor.AstLookupVisitor):
        def visit_integer(self, node):
            pass

    int42 = ast.Integer(42, None)
    lookup_visitor = MyAstLookupVisitor(ast.AstNodeType.INTEGER)
    int42.accept(lookup_visitor)

    # although the visitor is collecting Integer nodes, get_nodes() should
    # be empty since `visit_integer` method is overriden
    assert len(lookup_visitor.get_nodes()) == 0
```

And the `assert` fails (size is 1). It means the overridden `visit_integer` Python method is never called, but this happens **whether I remove the `visit_` methods in pyvisitor.cpp or not**.

IMO this is an issue that should be addressed in a separate PR.